test-suite-kotlin/src/test/kotlin/io/micronaut/http/client/SuspendClientSpec.kt

```diff
@@ -42,4 +43,14 @@ class SuspendClientSpec {
         Assertions.assertNull(response)
     }
 
+    @Test
+    fun testFlowAsPublisherInFilterStep() {
+        val server = ApplicationContext.run(EmbeddedServer::class.java, mapOf("spec.name" to "SuspendClientSpec"))
+        val ctx = server.applicationContext
+        val response = runBlocking {
+            ctx.getBean(SuspendClient::class.java).call(SuspendClientFilter.filterCheck)
+        }
+        Assertions.assertEquals(response, "{\"newState\":\"${SuspendClientFilter.filterCheck}\"}")
```

Comment:
we should close the server at the end of the test with `server.close()`

Reply:
I added a [commit to close the server](https://github.com/micronaut-projects/micronaut-core/pull/8715/commits/09b140d38ef3ff5fae5a3bfb130d038a6c8f4395)